storage/store.go

```diff
@@ -1855,6 +1855,27 @@ func (s *Store) enqueueRaftMessage(req *RaftMessageRequest) error {
 // Replica. It requires that processRaftMu is held and that s.mu is
 // not held.
 func (s *Store) handleRaftMessage(req *RaftMessageRequest) error {
+	// Drop messages that come from a node that we believe was once
+	// a member of the group but has been removed.
+	s.mu.Lock()
+	r, ok := s.mu.replicas[req.GroupID]
+	s.mu.Unlock()
+	if ok {
+		found := false
+		desc := r.Desc()
+		for _, rep := range desc.Replicas {
+			if rep.ReplicaID == req.FromReplica.ReplicaID {
+				found = true
+				break
+			}
+		}
+		// It's not a current member of the group. Is it from the future
+		// or the past?
+		if !found && req.FromReplica.ReplicaID < desc.NextReplicaID {
+			return nil
```

Comment:
Perhaps add a v2 level info message here about how the message is being dropped.